fix: support `configDir` for other `compilerOptions` fields

trikadin opened the pull request:
This PR adds support for the "${configDir}" variable interpolation for all the fields in "compilerOptions" object as listed in the [original issue](https://github.com/microsoft/TypeScript/issues/57485#issuecomment-2027787456)

trikadin commented:
Sorry I didn't open an issue, it was faster to make a fix than a repro example :)

privatenumber commented:
No problem, thanks for the PR!

Can you add a test or update the existing one for confibDir to include these properties?

trikadin commented:
> No problem, thanks for the PR!
> 
> Can you add a test or update the existing one for configDir to include these properties?

Added test cases. Also, I fixed a bug when "declarationDir" was missed in generated "excludes", now that matches behaviour of TS :)

privatenumber commented:
Ah nice catch! Thanks for the PR 🙌

privatenumber commented:
:tada: This PR is included in version 4.8.1 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/get-tsconfig/v/4.8.1)
- [GitHub release](https://github.com/privatenumber/get-tsconfig/releases/tag/v4.8.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
